### DISCUSSION TITLE
My Quasar CLI project with Vite build process takes a long time (hours)

### DISCUSSION THREAD

**Discussion by ricar2ce** [quasarframework/quasar#16953] [General - Components / Directives / etc] [answered]
```
quasar build

 .d88888b.
d88P" "Y88b
888     888
888     888 888  888  8888b.  .d8888b   8888b.  888d888
888     888 888  888     "88b 88K          "88b 888P"
888 Y8b 888 888  888 .d888888 "Y8888b. .d888888 888
Y88b.Y8b88P Y88b 888 888  888      X88 888  888 888
 "Y888888"   "Y88888 "Y888888  88888P' "Y888888 888
       Y8b


 Build mode............. spa
 Pkg quasar............. v2.14.6
 Pkg @quasar/app-vite... v1.7.3
 Pkg vite............... v2.9.17
 Debugging.............. no
 Publishing............. no

 App • Running "@quasar/apollo" Quasar App Extension...
 App • Extension(@quasar/apollo): Extending quasar.config.js...
 App •  WAIT  • Compiling of SPA UI with Vite in progress...
```

**After that it doesn't do anything else and in the dist/spa folder only the icons are created
any help please?**

**Comment by rstoenescu** [comment]
Hi,

I don't think anyone will be able to help you or at least guide you to find the culprit. Please provide a reproduction repo with clear instructions on how to reproduce.

**Reply by ricar2ce** [reply]
The build was already done, but it was after a long time (hours), is it normal for it to take so long in large projects?

**Reply by rstoenescu** [reply]
Definitely not. There‘s something in your specific case that has a bug. Could be a Vue plugin, a Vite plugin, something in your code or even the machine that you are building on. Would gladly help you, but I need your project to investigate with, otherwise we‘re just playing the guessing game. If you‘d like, you can contact me on Discord directly to find a way.

**Reply by ricar2ce** [reply]
and there is no way for me to see the details of what is happening during the build process so I can detect what is causing the error?
something similar to when the version 1 build process was run

**Reply by mr-Sepi0l** [reply]
> Hi,
> 
> I don't think anyone will be able to help you or at least guide you to find the culprit. Please provide a reproduction repo with clear instructions on how to reproduce.

Hey sorry to burry out this post, but I actually have the same issue, my build are taking ages. Not 3hrs, but a significant time since I added capacitor. Each time I'm building in spa mode to be sure it doesn't buld for the other platform but it takes really way longer than before.
And another issue that suddenly appears a while ago is my `quasar dev` has been slow as well and my interface not that reactive anymore.

I'll provide you an acces to a private repo if you're up to help ^^ Thanks

**Reply by yusufkandemir** [reply]
@mr-Sepi0l please see https://github.com/quasarframework/quasar/discussions/16953#discussioncomment-10982524 to hopefully debug your problem

**Comment by ricar2ce** [comment]
> Definitely not. There‘s something in your specific case that has a bug. Could be a Vue plugin, a Vite plugin, something in your code or even the machine that you are building on. Would gladly help you, but I need your project to investigate with, otherwise we‘re just playing the guessing game. If you‘d like, you can contact me on Discord directly to find a way.

I wrote to you on discord

**Comment by ricar2ce** [comment]
Please, I still have the problem, the build process always takes 3 hours, help please 🙏

**Comment by ricar2ce** [comment]
Can someone help me with an option to debug the build process? I need to detect what is causing the build to take more than 3 hours.

quasar dev --debug: don' t show anything

HELP PLEASE!!! 🙏

**Comment by yusufkandemir** [comment]
You can try `DEBUG='*' quasar build --debug`. Passing the `DEBUG='*'` env variable will make any underlying library, e.g. Vite, esbuild, etc., log every single detail they can. It might reveal some hints. But, without reproduction, we can't provide further help.

**Comment by ricar2ce** [comment, marked as answer]
Thank you very much, @yusufkandemir , your answer guided me in the right direction.
In my case, I use Windows, in Powershell you can use the following command:
```powershell
$env:DEBUG="quasar:*"; quasar build --debug
```
This gave me the possibility to see that the problem was being caused by the web-fonts of my project, specifically nucleo-icons and another custom one created with nucleo-icons; although when I removed them from the project, it also stopped for hours with line-awesome or the material design ones, so it was any icon library and the message in the logs was the following:

  ```
...
vite:config   }
  vite:config } +0ms
Browserslist: caniuse-lite is outdated. Please run:
  vite:resolve 3.22ms ./web-font/KFOlCnqEu92Fr1MmYUtfBBc-.woff -> C:/NodeJS/PROJECTS/envios-front/node_modules/@quasar/extras/roboto-font/web-font/KFOlCnqEu92Fr1MmYUtfBBc-.woff +0ms
  vite:resolve 4.95ms quasar/src/css/variables.sass -> C:/NodeJS/PROJECTS/envios-front/node_modules/quasar/src/css/variables.sass +212ms
  vite:resolve 0.36ms ./la-regular-400.woff -> C:/NodeJS/PROJECTS/envios-front/node_modules/@quasar/extras/line-awesome/la-regular-400.woff +3s
  vite:resolve 0.33ms ./la-brands-400.woff -> C:/NodeJS/PROJECTS/envios-front/node_modules/@quasar/extras/line-awesome/la-brands-400.woff +1ms
  vite:resolve 0.29ms ./la-solid-900.woff -> C:/NodeJS/PROJECTS/envios-front/node_modules/@quasar/extras/line-awesome/la-solid-900.woff +0ms

(!) Some chunks are larger than 500 KiB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/guide/en/#outputmanualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
 App •  DONE  • SPA UI compiled with success • 8492627ms
```

Solution:
The main solution was to adjust the chunk size warning limit:
```js
chunkSizeWarningLimit: 1000, // Increase the limit to 1 MB
```
Although I also added the following plugins in quasar.config.js:
```js
build: {
  // ... others configurations

  vueRouterMode: 'hash',
  extendViteConf(viteConf) {
    viteConf.build.rollupOptions = {
      output: {
        manualChunks(id) {
          if (id.includes('node_modules')) {
            return 'vendor'; // Separa las dependencias externas en un chunk separado
          }
        }
      }
    };
  },
  chunkSizeWarningLimit: 500,

  // ... others configurations

  vitePlugins: [
    {
      name: 'ignore-fonts',
      enforce: 'post',
      resolveId(id) {
        if (/\.(woff|woff2|ttf|eot|svg)$/.test(id)) return id;
        return null;
      },
      load(id) {
        if (/\.(woff|woff2|ttf|eot|svg)$/.test(id)) return '';
        return null;
      }
    }
  ]
}
```
